draft-ietf-quic-transport.md

```diff
@@ -1438,6 +1438,31 @@ New transport parameters can be registered according to the rules in
 {{iana-transport-parameters}}.
 
 
+## Cryptographic Message Buffering
+
+Implementations need to maintain a buffer of CRYPTO data received out of order.
+Because there is no flow control of CRYPTO frames, an endpoint could
+potentially force its peer to buffer an unbounded amount of data.
+
+Implementations MUST support buffering at least 4096 bytes of data received in
+CRYPTO frames out of order. Endpoints MAY choose to allow more data to be
+buffered during the handshake. A larger limit during the handshake could allow
+for larger keys or credentials to be exchanged. An endpoint's buffer size does
+not need to remain constant during the life of the connection.
+
+Being unable to buffer CRYPTO frames during the handshake leads to a connection
+failure. If an endpoint's buffer is exceeded during the handshake, it can expand
+its buffer temporarily to complete the handshake. If an endpoint cannot expand
+its buffer, it MUST close the connection with a CRYPTO_BUFFER_EXCEEDED error
+code.
+
+Once the handshake completes, if an endpoint is unable to buffer all data in a
+CRYPTO frame, it MAY discard all subsequent CRYPTO frames, or it MAY close the
```

Comment:
This isn't clear to me. If the endpoint is unable to buffer data in a CRYPTO frame, why does it need to drop _subsequent_ CRYPTO frames?

Reply:
Because TLS handshake messages require a stream that guarantees in-order delivery. Once you discard some bytes, you cannot use subsequent data on the stream. Therefore, the only way to keep the connection open after discarding some bytes is to discard all the following bytes, while sending ACKs.

Reply:
Ah, ok, thanks @kazuho.